### PR TITLE
#50 clear button

### PR DESCRIPTION
**Pull request by KatharinaBlass**
# Ticket
[#50](https://github.com/hpi-swa-teaching/SWT18-Project-08/issues/50)

# Description
Pressing the new clear-styles-button after selecting a piece of text will remove color and styles (bold, italic,...) from it.

# Is everything tested?
one test, which tests whether the run of textArea is empty after pressing the button (so no style is currently applied)

# Is it release relevant?
I guess so

# Screenshot
nothing being cool enough for a screenshot

# other checks
 - [x] no external ressources needed
 - [x] methods are formatted correctly
    - [x] no `.` at the end of your function
    - [x] functions have an empty line between functionname and code (except accessors)
    - [x] use cascades wherever possible
    - [x] variable definitions...
      - [ ] ... have an space between before/after the `|` symbol
      - [ ] ... have a blank line above and under
      - [ ] ... are used as few as possible
   
   ```smalltalk
   example: aValue
   
   | variable |
   
   self myObject 
       value: aValue;
       color: Color red.
   self otherMethod
   ```
   - [x] static values are stored on class Side
   - [ ] every function has a category
 - [x] basic UX
   - [ ] UI is intuitive to use (tested by the reviewer! reject if not)
   - [ ] ...

### CODE REVIEW

File: packages/RichText-Core.package/RTEAddComponent.class/instance/exportMarkdown.st [hunk 1]
```diff
@@ -1,6 +1,4 @@
 api
 exportMarkdown
 
-	(self coreObject notNil)
-		ifTrue: [^ self coreObject exportMarkdown]
-		ifFalse: [^ '']
+	^ self coreObject exportMarkdown
```
Comment (adrianjost) [inline packages/RichText-Core.package/RTEAddComponent.class/instance/exportMarkdown.st]: I think you made something wrong when merging. exportMarkdown should never return "nil" instead return an empty string. This was a fix and you remove it here. We should consider writing a test for this edge case. Could you do this?
It's just ... at the AddComponent Tests
```smalltalk
testExportMarkdownWhenNoComponentChosen

    self assert: component exportMarkdown equals: ''
```

File: packages/RichText-Core.package/RTELayoutComponent.class/instance/exportMarkdown.st [hunk 2]
```diff
@@ -5,5 +5,5 @@ exportMarkdown
 	
 	exportString := ''.
 	self submorphs do: [:submorph | 
-		exportString := exportString, (submorph exportMarkdown), (Character cr)].
+		exportString := exportString, (submorph exportMarkdown), (Character cr), (Character cr)].
```
Comment (adrianjost) [inline packages/RichText-Core.package/RTELayoutComponent.class/instance/exportMarkdown.st]: wrong version here to. old one is correct

File: packages/RichText-Core.package/RichTextEditor.class/instance/initialize.st [hunk 3]
```diff
@@ -1,14 +1,11 @@
...
 	self
 		extent: 500@200;
 		
 		flag: #refactor;
 		"adding the addComponent should be handled by the Layout component itself!"
-		coreObject: (RTEVerticalLayoutComponent new addComponent: RTEAddComponent).
-	wrapper := self initializeWrapper.
-	self content: wrapper
\ No newline at end of file
+		coreObject: (RTEVerticalLayoutComponent new addComponent: RTEAddComponent);
```
Comment (adrianjost) [inline packages/RichText-Core.package/RichTextEditor.class/instance/initialize.st]: and here. I think you removed almost any new feature :D

Comment (adrianjost) [reply packages/RichText-Core.package/RichTextEditor.class/instance/initialize.st]: I think you should undo the last commit and merge again. There is so much wrong here
